src/components/tab-nav/tab-nav.tsx

```diff
@@ -370,9 +375,42 @@ export class TabNav {
   }
 
   get enabledTabTitles(): HTMLCalciteTabTitleElement[] {
-    return filterDirectChildren<HTMLCalciteTabTitleElement>(
+    const directEnabled = filterDirectChildren<HTMLCalciteTabTitleElement>(
       this.el,
       "calcite-tab-title:not([disabled])"
     );
+    return directEnabled.filter((tabTitle) => !tabTitle.closed);
+  }
+
+  handleTabTitleClose(closedTabTitleEl: HTMLCalciteTabTitleElement): void {
+    const { tabTitles } = this;
+
+    const visibleTabTitlesIndices = tabTitles.reduce((acc, curr, i) => {
```

Comment:
I think this method can be simplified a bit further, but let's sync up on this tomorrow.